api/mapping/views.py

```diff
@@ -198,13 +200,18 @@ def form_valid(self, form):
             pk=self.kwargs.get('pk')
         )
 
-        mapping = MappingRule.objects.create(
+        mapping,created = MappingRule.objects.get_or_create(
```

Comment:
get or create returns a tuple mapping, created where created is a boolean value. If created is True means the mapping is unique so it saves it. If created is False then that means it does not create a new object since it already exists.